edb/schema/rewrites.py

```diff
@@ -0,0 +1,301 @@
+            # parent_vname = source.get_verbosename(schema)
+            # pol_name = self.get_verbosename(parent=parent_vname)
+            # in_ddl_context_name = pol_name
+
+            kind = self._get_kind(schema)
+
+            anchors = {}
+            if qltypes.RewriteKind.Update == kind:
+                anchors['__old__'] = pathid.PathId.from_type(
+                    schema,
+                    source,
+                    typename=sn.QualName(module='__derived__', name='__old__'),
+                )
+
+            anchors['__specified__'] = pathid.PathId.from_type(
+                schema,
+                source,
+                typename=sn.QualName(
+                    module='__derived__', name='__specified__'
+                ),
+            )
+
+            singletons = frozenset(anchors.values())
+
+            assert isinstance(source, s_types.Type)
+
+            return type(value).compiled(
+                value,
+                schema=schema,
+                options=qlcompiler.CompilerOptions(
+                    modaliases=context.modaliases,
+                    schema_object_context=self.get_schema_metaclass(),
+                    anchors=anchors,
+                    singletons=singletons,
+                    apply_query_rewrites=not context.stdmode,
+                    track_schema_ref_exprs=track_schema_ref_exprs,
+                    # in_ddl_context_name=in_ddl_context_name,
+                    detached=True,
+                ),
+            )
+        else:
+            return super().compile_expr_field(
+                schema, context, field, value, track_schema_ref_exprs
+            )
+
+    def get_dummy_expr_field_value(
+        self,
+        schema: s_schema.Schema,
+        context: sd.CommandContext,
+        field: so.Field[Any],
+        value: Any,
+    ) -> Optional[s_expr.Expression]:
+        if field.name == 'expr':
+            return s_expr.Expression(text='false')
+        else:
+            raise NotImplementedError(f'unhandled field {field.name!r}')
+
+    def validate_object(
+        self,
+        schema: s_schema.Schema,
+        context: sd.CommandContext,
+    ) -> None:
+        # XXX: verify we don't have the same bug as access policies
+        # where linkprop defaults are broken.
+        # (I think we won't need to, since we'll operate after
+        # the *real* operations)
+        pass
+
+    @classmethod
+    def _cmd_tree_from_ast(
+        cls,
+        schema: s_schema.Schema,
+        astnode: qlast.DDLOperation,
+        context: sd.CommandContext,
+    ) -> sd.Command:
+        group = sd.CommandGroup()
+
+        assert isinstance(astnode, qlast.RewriteCommand)
+
+        for kind in astnode.kinds:
+            cmd = super()._cmd_tree_from_ast(schema, astnode, context)
+            assert isinstance(cmd, RewriteCommand)
+
+            cmd.set_attribute_value('kind', kind)
+            group.add(cmd)
+        return group
```

Comment:
@msullivan This is the part I need you to checkout.

Reply:
Yeah, this seems like a good way to handle it, but I'd add a comment explaining it both here and in CreateRewrite